src/cryptopp-1-fixes.patch

```diff
@@ -0,0 +1,17 @@
+This file is part of MXE.
+See index.html for further information.
+
+Fix broken Makefile, add all relevant cpp files
+
+diff -ur a/GNUmakefile b/GNUmakefile
+--- a/GNUmakefile	2015-11-22 22:17:59.000000000 +0100
++++ b/GNUmakefile	2015-11-26 15:32:40.617516683 +0100
+@@ -255,7 +255,7 @@
+ LIBOBJS := $(filter-out $(TESTOBJS),$(OBJS))
+ 
+ # List cryptlib.cpp first in an attempt to tame C++ static initialization problems
+-DLLSRCS := cryptlib.cpp algebra.cpp algparam.cpp asn.cpp basecode.cpp cbcmac.cpp channels.cpp des.cpp dessp.cpp dh.cpp dll.cpp dsa.cpp ec2n.cpp eccrypto.cpp ecp.cpp eprecomp.cpp files.cpp filters.cpp fips140.cpp fipstest.cpp gf2n.cpp gfpcrypt.cpp hex.cpp hmac.cpp integer.cpp iterhash.cpp misc.cpp modes.cpp modexppc.cpp mqueue.cpp nbtheory.cpp oaep.cpp osrng.cpp pch.cpp pkcspad.cpp pubkey.cpp queue.cpp randpool.cpp rdtables.cpp rijndael.cpp rng.cpp rsa.cpp sha.cpp simple.cpp skipjack.cpp strciphr.cpp trdlocal.cpp
++DLLSRCS += 3way.o adhoc.o adler32.o algebra.o algparam.o arc4.o asn.o authenc.o base32.o base64.o basecode.o bench2.o bench.o bfinit.o blowfish.o blumshub.o camellia.o cast.o casts.o cbcmac.o ccm.o channels.o cmac.o cpu.o crc.o cryptlib.o datatest.o default.o des.o dessp.o dh2.o dh.o dll.o dlltest.o dsa.o eax.o ec2n.o eccrypto.o ecp.o elgamal.o emsa2.o eprecomp.o esign.o files.o filters.o fips140.o fipsalgt.o fipstest.o gcm.o gf2_32.o gf256.o gf2n.o gfpcrypt.o gost.o gzip.o hex.o hmac.o hrtimer.o ida.o idea.o integer.o iterhash.o luc.o mars.o marss.o md2.o md4.o md5.o misc.o modes.o mqueue.o mqv.o nbtheory.o network.o oaep.o osrng.o panama.o pkcspad.o polynomi.o pssr.o pubkey.o queue.o rabin.o randpool.o rc2.o rc5.o rc6.o rdrand.o rdtables.o regtest.o rijndael.o ripemd.o rng.o rsa.o rw.o safer.o salsa.o seal.o seed.o serpent.o sha3.o shacal2.o sha.o sharkbox.o shark.o skipjack.o socketft.o sosemanuk.o square.o squaretb.o strciphr.o tea.o test.o tftables.o tiger.o tigertab.o trdlocal.o ttmac.o twofish.o validat1.o validat2.o validat3.o vmac.o wait.o wake.o whrlpool.o winpipes.o xtrcrypt.o xtr.o zdeflate.o zinflate.o zlib.o
+ DLLOBJS := $(DLLSRCS:.cpp=.export.o)
+ 
```

Comment:
Thank your for spotting this!  After fiddling around for quite a while, I realize that it just doesn't work if the %.export.o rule is used.  It seems -DCRYPTOPP_EXPORT, what is then defined, does some weird (MSVC I guess) magic that keeps things from working.  I even tested the resulting DLL file with the test binary under windows, it works perfectly fine without the export rule.  So this is actually what should be set, and I updated the patch to reflect this.